Mathlib/SetTheory/Game/Impartial.lean

```diff
@@ -47,18 +46,20 @@ theorem impartial_def {G : PGame} :
 
 namespace Impartial
 
-instance impartial_zero : Impartial 0 := by rw [impartial_def]; dsimp; simp
+instance impartial_zero : Impartial 0 := by
+  rw [impartial_def]
+  simp
```

Comment:
I wonder if `simp [impartial_def]` works as the proof here, and below?

Reply:
No, since `impartial_def` refers to `Impartial` this just causes an infinite recursion.